Sources/ClientRuntime/Networking/Http/SdkHttpRequest.swift

```diff
@@ -83,6 +89,28 @@ extension SdkHttpRequest {
         httpRequest.body = nil
         return httpRequest
     }
+
+    public func toURLRequest() async throws -> URLRequest {
+        // Set URL
+        guard let url = self.endpoint.url else {
+            throw ClientError.dataNotFound("Failed to construct URLRequest due to missing URL.")
+        }
+        var request = URLRequest(url: url)
+        // Set method type
+        request.httpMethod = self.method.rawValue
+        // Set body
+        guard let body = try await self.body.readData() else {
```

Comment:
`URLRequest` does have a `httpBodyStream` property for streaming the request body.  But for now, reading the entire body into memory should be sufficient.  If customers want to use presigned requests with bodies to large for RAM, we can revisit this.

Reply:
Gotcha